src/js/route.js

```diff
@@ -1,267 +1,218 @@
+
+        if (Array.isArray(params)) {
+            // If the parameters are an array they have to be in order, so we can transform them into
+            // an object by keying them with the template segments in the order they appear
+            params = params.reduce((result, current, i) => ({ ...result, [segments[i].name]: current }), defaults);
+        } else if (segments.length === 1 && !params[segments[0].name] && params[Object.values(route.bindings)[0]]) {
+            // If there is only one parameter template segment and `params` is an object, that
+            // object is ambiguous—it could be the parameter key and value, or it could be an
+            // object representing just the parameter value; we can inspect it to find out,
+            // and if it's the parameter value, wrap it in an object with its key
+            params = { [segments[0].name]: params };
         }
 
-        let remainingParams = Object.assign(this.urlParams, this.queryParams);
-
-        return stringify(remainingParams, {
-            encodeValuesOnly: true,
-            skipNulls: true,
-            addQueryPrefix: true,
-            arrayFormat: 'indices'
-        });
-    }
-
-    current(name = null) {
-        let routeNames = Object.keys(this.ziggy.namedRoutes);
+        // Substitute any registered route model bindings for this route
+        return Object.entries({ ...defaults, ...params }).reduce((result, [key, value]) => {
+            const bound = value && typeof value === 'object' && route.bindings[key];
 
-        let currentRoute = routeNames.filter(name => {
-            if (this.ziggy.namedRoutes[name].methods.indexOf('GET') === -1) {
-                return false;
+            if (bound && !value.hasOwnProperty(route.bindings[key])) {
+                throw new Error(`Ziggy error: object passed as '${key}' parameter is missing route model binding key '${route.bindings[key]}'.`)
             }
 
-            return new Router(
-                name,
-                undefined,
-                undefined,
-                this.ziggy
-            ).matchUrl();
-        })[0];
+            return { ...result, [key]: bound ? value[route.bindings[key]] : value };
+        }, {})
+    }
 
-        if (name) {
-            const pattern = new RegExp(
-                '^' + name.replace('.', '\\.').replace('*', '.*') + '$',
-                'i'
-            );
-            return pattern.test(currentRoute);
+    // Get the parameter values from the current window URL, based on the given route definition
+    _dehydrate(route) {
+        let pathname = window.location.pathname
+            .replace(this._config.baseUrl.replace(/^\w*:\/\/[^/]+/, ''), '') // Remove subdirectories
+            .replace(/^\/+/, '');
+
+        // Given the hydrated string, template, and delimiter, extract and return
+        // an object of parameter names and values from part of a path or URL
```

Comment:
```suggestion
        // Given a URL piece, a template, and a delimiter, extract and return
        // an object of parameter names and values
```

I don't love "URL piece" but hydrated doesn't follow for a first timer to the codebase